File: clojure/space-age/src/space_age.clj
(ns space-age)

(def orbital-seconds
  {:earth 31557600
   :mercury 7600543
   :venus 19414149
   :mars 59354032
   :jupiter 374355659
   :saturn 929292362
   :uranus 2651370019
   :neptune 5200418560})

(defn on-earth [seconds]
  (/ seconds (:earth orbital-seconds)))

(defn on-mercury [seconds]
  (/ seconds (:mercury orbital-seconds)))

(defn on-venus [seconds]
  (/ seconds (:venus orbital-seconds)))

(defn on-mars [seconds]
  (/ seconds (:mars orbital-seconds)))

(defn on-jupiter [seconds]
  (/ seconds (:jupiter orbital-seconds)))

(defn on-saturn [seconds]
  (/ seconds (:saturn orbital-seconds)))

(defn on-neptune [seconds]
  (/ seconds (:neptune orbital-seconds)))

(defn on-uranus [seconds]
  (/ seconds (:uranus orbital-seconds)))

File: go/space-age/space_age.go
package space

type Planet string

func Age(seconds float64, planet Planet) float64 {
	switch planet {
	case "Earth":
		return seconds / 31_557_600
	case "Mercury":
		return seconds / 7_600_543
	case "Venus":
		return seconds / 19_414_149
	case "Mars":
		return seconds / 59_354_032
	case "Jupiter":
		return seconds / 374_355_659
	case "Saturn":
		return seconds / 929_292_362
	case "Uranus":
		return seconds / 2_651_370_019
	case "Neptune":
		return seconds / 5_200_418_560
	default:
		return 0
	}
}
